packages/terra-paginator/src/ControlledProgressivePaginator.jsx

```diff
@@ -81,8 +86,14 @@ class ControlledProgressivePaginator extends React.Component {
     return (
       <div className={cx('paginator', 'progressive', theme.className)} role="navigation" aria-label="pagination">
         <div>
-          {(totalCount) ? intl.formatMessage({ id: 'Terra.paginator.pageCount' }, { pageNumber: selectedPage, pageNumberTotal: totalPages })
-            : intl.formatMessage({ id: 'Terra.paginator.pageIndex' }, { pageNumber: selectedPage })}
+          {
+            /* eslint-disable no-nested-ternary */
+            (totalCount && pageLabel) ? intl.formatMessage({ id: 'Terra.paginator.pageCountWithLabel' }, { pageLabel, pageNumber: selectedPage, pageNumberTotal: totalPages })
+              : (totalCount) ? intl.formatMessage({ id: 'Terra.paginator.pageCount' }, { pageNumber: selectedPage, pageNumberTotal: totalPages })
+                : (pageLabel) ? intl.formatMessage({ id: 'Terra.paginator.pageIndexWithLabel' }, { pageLabel, pageNumber: selectedPage })
+                  : intl.formatMessage({ id: 'Terra.paginator.pageIndex' }, { pageNumber: selectedPage })
```

Comment:
Not sure if this is any simplier, but might be easier to read through:
```js
let messageID;
let messageAttributes;

if (totalCount) {
  if (pageLabel) {
     messageID = 'Terra.paginator.pageCountWithLabel'
     messageAttributes = { pageLabel, pageNumber: selectedPage, pageNumberTotal: totalPages }
  } else {
..
  }
} else {
...
}

return (
      <div className={cx('paginator', 'progressive', theme.className)} role="navigation" aria-label="pagination">
        <div>
           intl.formatMessage({ id: messageId }, messageAttributes)

```

Reply:
Updated here https://github.com/cerner/terra-core/pull/3470/commits/839d26620a815100d31caab980928c3c9755ea57